pkg/xray/provider_test.go

```diff
@@ -42,16 +44,82 @@ func getTestResty(t *testing.T) *resty.Client {
 }
 
 func testAccPreCheck(t *testing.T) {
-	restyClient := getTestResty(t)
-	// TODO check the payload and make sure it's the right license type
-	_, err := restyClient.R().Get("/artifactory/api/system/licenses/")
-	if err != nil {
-		t.Fatal(err)
-	}
 	ctx := context.Background()
-	provider, _ := testAccProviders["xray"]()
+	provider, _ := testAccProviders()["xray"]()
 	oldErr := provider.Configure(ctx, terraform.NewResourceConfigRaw(nil))
 	if oldErr != nil {
 		t.Fatal(oldErr)
 	}
 }
+
+// Create a local repository with Xray indexing enabled. It will be used in the tests
+func testAccCreateRepos(t *testing.T, repos []string) {
+	restyClient := getTestResty(t)
+
+	type Repository struct {
+		Rclass    string `json:"rclass"`
+		XrayIndex bool   `json:"xrayIndex"`
+	}
+
+	repository := Repository{}
+	repository.Rclass = "local"
+	repository.XrayIndex = true
+	for _, repo := range repos {
+		response, errRepo := restyClient.R().SetBody(repository).Put("artifactory/api/repositories/" + repo)
+		repoExists := strings.Contains(fmt.Sprint(errRepo), "Case insensitive repository key already exists")
```

Comment:
Let's add a comment here, documenting why we aren't checking for status code and check for string in response for this condition.

Reply:
added